pkg/gardenadm/cmd/init/init.go

```diff
@@ -60,6 +62,13 @@ func run(ctx context.Context, opts *Options) error {
 		return fmt.Errorf("failed bootstrapping control plane: %w", err)
 	}
 
+	if err := b.FS.MkdirAll(filepath.Dir(cmd.ConfigDirLocation), os.ModeDir); err != nil {
+		return fmt.Errorf("failed config directory location file %s: %w", cmd.ConfigDirLocation, err)
+	}
+	if err := b.FS.WriteFile(cmd.ConfigDirLocation, []byte(opts.ConfigDir), 0644); err != nil {
+		return fmt.Errorf("failed writing config directory location file %s: %w", cmd.ConfigDirLocation, err)
+	}
```

Comment:
We basically create a poor man's symbolic link. Having a real symbolic link from `/etc` to a random place is probably not what we want, but this still feels a bit weird. I have no better idea, though.